src/index.py

```diff
@@ -32,13 +43,35 @@ def play(grid_arg, started_grid):
         grid, shuffled = play_one_turn(grid, shuffled)
 
 
+def change_size(min_size, max_size, default_size):
+    if input(show_menu_size('W')) in ['w', 'W']:
+        size_valid = False
+        while not size_valid:
+            try:
+                size = int(input(ask_size()))
+                if not min_size <= size <= max_size:
```

Comment:
Duplicated code from build grid. You better have to check for invalid size exception